src/lib/BasicSide.tsx

```diff
@@ -1,92 +1,44 @@
-import React, { useState } from 'react'
-
 import { useSelector } from 'react-redux'
 
 import { useNavigate, useLocation } from 'react-router-dom'
 
-import {
-  AppBar,
-  Box,
-  Button,
-  ButtonGroup,
-  ToggleButton,
-  ToggleButtonGroup,
-  Container,
-  CssBaseline,
-  Divider,
-  Grid,
-  List,
-  ListItem,
-  ListItemButton,
-  ListItemIcon,
-  ListItemText,
-  Toolbar,
-  Typography,
-  IconButton
-} from '@mui/material'
-
-import {
-  ChevronLeft,
-  MoveToInbox as InboxIcon,
-  Mail as MailIcon,
-
-  FactoryOutlined,
-  KeyOutlined,
-  AssignmentTurnedInOutlined,
-  TextSnippetOutlined,
-  HighlightAlt,
-  SupervisorAccount as AccountIcon,
-  Map as MapIcon,
-  Tablet as TabletIcon,
-  Update as UpdateIcon,
-  Security as AdminIcon,
-  ContentPaste as ClipBoardIcon,
-  FitScreen,
-  Apps as DotsSquareIcon
-
-} from '@mui/icons-material'
-
-import {
-  BasicDrawer,
-  BasicDrawerHeader
-} from './BasicDrawer'
-import { Exact, Gubu } from 'gubu'
-
-const iconmap: any = {
-  factory: FactoryOutlined,
-  key: KeyOutlined,
-  done: AssignmentTurnedInOutlined,
-  docs: TextSnippetOutlined,
-  hightlight: HighlightAlt,
-  map: MapIcon,
-  account: AccountIcon,
-  tablet: TabletIcon,
-  update: UpdateIcon,
-  admin: AdminIcon,
-  clipboard: ClipBoardIcon,
-  fitscreen: FitScreen,
-  'dots-square': DotsSquareIcon
-}
-
-function makeIcon(name: string) {
-  let Icon = iconmap[name]
-  return <Icon />
-}
+import { Child, Gubu, Open, Required } from 'gubu'
+import BasicSideMenu from './BasicSideMenu'
+import { ChevronLeft } from '@mui/icons-material'
+import { Divider, IconButton } from '@mui/material'
+import { BasicDrawer, BasicDrawerHeader } from './BasicDrawer'
+
+// spec schema definition with Gubu
+const BasicSideSpecShape = Gubu(Required({
+  side: {
+    logo: {
+      img: String
+    },
+    section: Child({
+      title: String,
+      item: Child({
+        kind: Number,
+        label: Number,
+        icon: String,
+        path: String,
+        access: Child(Boolean)
```

Comment:
use Child(Boolean,{}) since access should be an open object

Reply:
same for item

Reply:
https://github.com/rjrodger/gubu/issues/7 doc needed - my task

Reply:
@rjrodger for item ,should I do this? 

```
item: Child({
        kind: Number,
        label: Number,
        icon: String,
        path: String,
        access: Child(Boolean, {})
      }. {})
```

Reply:
Also I updated `Child(Boolean,{})`